test/utils/utils.test.js

```diff
@@ -329,4 +329,16 @@ describe('Utils', () => {
     });
     expect(io instanceof IntersectionObserver).to.be.true;
   });
+
+  it('adds privacy trigger to cookie preferences link in footer', () => {
+    window.adobePrivacy = { showPreferenceCenter: sinon.spy() };
+    if (!document.querySelector('footer')) {
+      const footerHtml = '<footer><a href="https://www.adobe.com/#openPrivacy" id="privacy-link">Cookie preferences</a></footer>';
+      document.body.insertAdjacentHTML('beforeEnd', footerHtml);
+    }
```

Comment:
Thanks for pressing on this. I think I came to a better solution.

Basically the problem was that the `document.body` was being added at the top of the file, being shared by all tests, but one test removed the footer. Since the test ran async, I was getting different results and fails per run.

To fix this, I wrapped all of the tests sharing the same `document.body` in a `describe` and separated out my test with its own `document.body`.

Reply:
Awesome! Glad you found a good solution!